Common/Data/Consolidators/BaseDataConsolidator.cs

```diff
@@ -73,7 +73,7 @@ protected override void AggregateBar(ref TradeBar workingBar, BaseData data)
                 workingBar = new TradeBar
                 {
                     Symbol = data.Symbol,
-                    Time = data.Time,
+                    Time = GetRoundedBarTime(data.Time),
```

Comment:
Absolutely wonderful, I was thinking of this same solution. Many thanks for the implementation!